enhancements/installer/aws-permissions-check-bypass.md

```diff
@@ -0,0 +1,307 @@
+
+## Motivation
+
+Presently, when installing OpenShift on AWS, the credentials provided to the
+installer are queried/checked for appropriate permissions to verify that:
+1. Can be used to complete an installation (the installer set of required permissions)
+1. The in-cluster cloud-credential-operator (CCO) can successfully operate in
+   either 'mint' or 'passthrough' mode.
+If any of these conditions are not met, the install will fail and explain which
+permissions were deemed to be missing.
+
+When an AWS account is configured with Service Control Policies (SCP) the
+permissions checking/simulation APIs can provide incorrect results depending on
+the contents of the SCPs. Service Control Policies are typically used to deny
+certain API calls unless a condition (or set of conditions) is met.  For
+example, if user `openshift` has a policy attached that allows `ec2:*`, but the
+SCP at the account level denies `ec2:*` unless the API calls are made against
+region `us-east-1`, then the `openshift` user would receive errors when making
+EC2 API calls outside of region `us-east-1`.
+
+Attempting to validate whether user `openshift` has permissions to perform
+`ec2:DescribeInstances` against region `us-east-1` will result in a
+determination that the user `openshift` cannot succesfully perform the API call,
+even though the actual `ec2:DescribeInstances` call works (as long as it is
+against the allowed region).
+
+In order to acomodate installing OpenShift in these environments, a way is
+needed for the individual performing the installation to indicate that these
+permissions checks should be skipped.
+
+Additionally, a mechanism is needed to indicate to the cloud-credential-operator
+to force it into either `mint` or `passthrough` mode, so that it too can avoid
+attempting to validatate permissions.
+
+### Goals
+
+Enable successfull installation and operation of OpenShift in these AWS accounts
+where the results of permissions simulations cannot be relied upon.
+
+### Non-Goals
+
+Not looking to write complex policy introspection to implement what should
+already be performed by the AWS permissions simulation API.
+
+## Proposal
+
+### Installer
+Introduce an install-config.yaml field that can be populated by users to
+indicate that the installer should not concern itself with determining whether
+the credentials provided are sufficient for installation and to affect the
+in-cluster behavior of cloud-credential-operator.
+
+Extend the install-config type in the installer repo:
+```
+type InstallConfig struct {
+	// ForceCredentialsMode instructs the installer to not attempt to query
+	// the cloud permissions before attempting installation. It also passes
+	// down the desired credentials mode to the cloud-credential-operator
+	// so that it too does not attempt to query permissions.
+	// +optional
+	ForceCredentialsMode cloudcredop.CloudCredentialsMode `json:"forceCredentialsMode,omitempty"`
```

Comment:
> https://github.com/openshift/installer/blob/master/docs/dev/explain.md#additional-guidelines-on-godoc-comments (1)

using cred minter's API in installconfig is against the guildelines for InstallConfig API.